frame/contracts/src/storage/meter.rs

```diff
@@ -560,74 +678,44 @@ mod tests {
 		)
 	}
 
-	#[test]
-	fn existential_deposit_works() {
-		clear_ext();
-
-		let mut meter = TestMeter::new(&ALICE, Some(1_000), 0).unwrap();
-		assert_eq!(meter.available(), 1_000);
-
-		// a `Refund` will be turned into a `Charge(ed)` which is intended behaviour
-		let mut nested0 = meter.nested();
-		nested0.charge(&Diff { require_ed: true, ..Default::default() }).unwrap();
-		nested0
-			.charge(&Diff { bytes_removed: 1, require_ed: true, ..Default::default() })
-			.unwrap();
-		meter.absorb(nested0, &ALICE, &BOB, None);
-
-		assert_eq!(
-			TestExtTestValue::get(),
-			TestExt {
-				limit_checks: vec![LimitCheck { origin: ALICE, limit: 1_000, min_leftover: 0 }],
-				charges: vec![Charge {
-					origin: ALICE,
-					contract: BOB,
-					amount: Deposit::Charge(<Test as Config>::Currency::minimum_balance() * 2),
-					terminated: false,
-				}]
-			}
-		)
-	}
-
 	#[test]
 	fn charging_works() {
 		clear_ext();
 
-		let min_balance = <Test as Config>::Currency::minimum_balance();
-
 		let mut meter = TestMeter::new(&ALICE, Some(1_000), 0).unwrap();
```

Comment:
```suggestion
		let mut meter = TestMeter::new(&ALICE, Some(100), 0).unwrap();
```

Reply:
this for checking that `RawMeter::charge()` really does not enforce the consuming deposit limit (see below), which is now done by `RawMeter::enforce_limit()`